internal/resources/helmcharts/helm_charts_data_source.go

```diff
@@ -46,3 +52,57 @@ var helmSchema = map[string]*schema.Schema{
 	},
 	spec.ChartsKey: spec.Charts,
 }
+
+func dataSourceHelmChartsRead(ctx context.Context, d *schema.ResourceData, m interface{}) (diags diag.Diagnostics) {
+	config := m.(authctx.TanzuContext)
+
+	metadataName, ok := d.Get(ChartMetadataNameKey).(string)
+	if !ok {
+		return diag.Errorf("Unable to read chart metadata name")
+	}
+
+	name, ok := d.Get(NameKey).(string)
+	if !ok {
+		return diag.Errorf("Unable to read chart name")
+	}
+
+	repositoryName, ok := d.Get(RepositoryNameKey).(string)
+	if !ok {
+		return diag.Errorf("Unable to read repository name")
+	}
+
+	resp, err := config.TMCConnection.OrganizationHelmChartsResourceService.VmwareTanzuManageV1alpha1ClusterFluxcdChartResourceServiceList(&chartsmodel.VmwareTanzuManageV1alpha1OrganizationFluxcdHelmRepositoryChartmetadataChartSearchScope{
+		Name:              name,
+		ChartMetadataName: metadataName,
+		RepositoryName:    repositoryName,
+	})
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	if len(resp.Charts) == 0 {
+		return diag.Errorf("Tanzu Mission Control no chart entry found with meatadata name : %s", metadataName)
```

Comment:
Can you rephrase the error message and also check metadata spelling.

Reply:
Updated PTAL